### ISSUE TITLE
"i" as a node name leads to compilation error

### ISSUE THREAD

**Issue by DRJP** [nimble-dev/nimble#1219]
Hi

The following model does not compile.  I'm using emacs, on Ubuntu and have the latest nimble from CRAN. 

```
code <- nimbleCode({
  i ~ dexp(1) 
})
rModel <- nimbleModel(code)
cModel <- compileNimble(rModel)
```

The model does compile if the node name is changed.  Switching to a deterministic node does not change anything. So the issue seems to be how nimble (or my compiler) is working with the node name "i". 

Here is the output on my screen. 

```
> cModel <- compileNimble(rModel)
Compiling
  [Note] This may take a minute.
  [Note] Use 'showCompilerOutput = TRUE' to see C++ compilation details.
Error: Failed to create the shared library. Run 'printErrors()' to see the compilation errors.
> printErrors()
P_1_code_MID_1.cpp: In member function ‘void code_MID_1::pointAtAll(code_MID_1_MV_UID_1*, int)’:
P_1_code_MID_1.cpp:52:3: error: invalid conversion from ‘NimArr<1, double>*’ to ‘int’ [-fpermissive]
   52 | i=&(values->i_Vec[i]);
make: *** [/usr/lib/R/etc/Makeconf:177: P_1_code_MID_1.o] Error 1
```
  
Best regards
David

**Comment by perrydv** [comment]
Thanks for this report @DRJP .  We'll take a look.

**Comment by paciorek** [comment]
We need to change the names of `i` and `values` as the arguments to `pointAtAll` so that we don't have overloaded use of variables in `pointAtAll` from the model variable and the argument names for `pointAtAll`. For now I'm doing this as `i_` and `values_`.

**Comment by paciorek** [comment]
Fixed by PR #1239
